printer-profile.1
'\" t
'\"
'\"	DO NOT EDIT! This file is generated from printer-profile.1in
'\"
.TH printer-profile 1 "Sun Apr 25 16:22:30 2021" "printer-profile 0.0"
'\"
'\"==========================================================================
'\"	STRINGS and MACROS
'\"==========================================================================
'\"
'\"	Define strings for special characters that nroff doesn't have
'\"
'\"	N.B. using .if cua to test for special characters did not work.
'\"
.if !'\*[.T]'ps' .ds ua up
.if  '\*[.T]'ps' .ds ua \(ua
.if !'\*[.T]'ps' .ds da down
.if  '\*[.T]'ps' .ds da \(da
.if !'\*[.T]'ps' .ds <- left
.if  '\*[.T]'ps' .ds <- \(<-
.if !'\*[.T]'ps' .ds -> right
.if  '\*[.T]'ps' .ds -> \(->
'\"
'\"     bx - box a Courier string for making keycaps
'\"
'\"	N.B. this mess is to make the line drawing come out only
'\"	if we are really generating postscript
'\"
.de bx
.ie !'\*[.T]'ps' \{\
.	RB [ \\$1 ]\\$2
.\}
.el \{\
.	ie !r ps4html \{\
'\" \(br\|\s-1\f(CB\\$1\fP\s+1\|\(br\l'|0\(rn'\l'|0\(ul'
.		ft CW
.		nr par*bxw \w'\\$1'+.4m 
\Z'\v'.25m'\D'l 0 -1m'\D'l \\n[par*bxw]u 0'\D'l 0 1m'\D'l -\\n[par*bxw]u 0''\
\Z'\h'.2m'\s-1\\$1\s+1'\
\h'\\n[par*bxw]u'\\$2
.		ft P
.	\}
.	el \{\
.		RB [ \\$1 ]\\$2
.	\}
.\}
..
'\"
'\" strings to set current color (null with old groff)
'\"
.if mred .ds red \m[red]
.if mgreen .ds green \m[green]
.if mblue .ds blue \m[blue]
.if mblack .ds black \m[black]
.if mblack .ds mP \mP
'\"
'\" fix for grotty + xterm. We call for orange, grotty outputs yellow,
'\" but xterm displays yellow as orange.  The cycle is complete.
'\"
.if n .defcolor orange rgb #ffff00
'\"
'\" color <color> - set the current color (ignores request with old groff)
'\"
.de color
.if mred \m[\\$1]\c
..
'\"
'\" colorword <color> <word> - colorize a word (ignored by old groff)
'\"
.de colorword
.ie m\\$1 \m[\\$1]\\$2\mP\c
.el \\$2\c
..
'\"
'\" colbox <fg> <bg> <word> - colorize a word in a filled box
'\"
.de colbox
.ie mred \M[\\$2]\
\v'+.167v'\
\D'P 0 -0.9v  \w'\\$3'u 0  0 +0.9v   -\w'\\$3'u 0'\
\v'-.167v'\
\m[\\$1]\\$3\mP\MP
.el \\$3\c
..
'\"
'\"	Macros for doing pdfmarks
'\"
.de specialps
.if  '\*[.T]'ps' \\k_\X'ps: \\$*'\h'|\\n_u'\c
..
'\"
'\" pdfmark PDFMARKCODE
'\"
.ds pdfmarks
.if d pdfmarks \{\
.de pdfmark
.	specialps exec [\\$1 pdfmark
..
'\"
'\" pdfdest LINKNAME
'\"
.de pdfdest
.pdfmark "/Dest /\\$1 /View [/XYZ -5 PL null] /DEST"
..
'\"
'\" pdfbookmark COUNT LINKNAME STRING
'\"
.de pdfbookmark
.   pdfmark "/View [/XYZ 44 730 1.0] /Count \\$1 /Dest /\\$2 /Title (\\$3) /OUT"
..
'\"
'\"	Define the SH and SS macros to save pdfmark information
'\"	in "arrays" of numbers and strings.
'\"
.if !r rr_n \{\
.nr rr_n 0 1
.am SH
.	nr rr_levels!\\n+[rr_n] 2
.	ds rr_labels!\\n[rr_n] \\$*
.	pdfdest Link\\n[rr_n]
..
.am SS
.	nr rr_levels!\\n+[rr_n] 3
.	ds rr_labels!\\n[rr_n] \\$*
.	pdfdest Link\\n[rr_n]
..
.\}
'\"
'\"	Called at the end of the document to generate the pdfmark outline
'\"
.de pdf_outline
.nr rr_levels!\\n+[rr_n] 1
.nr rr_i 0 1
.while \\n+[rr_i]<\\n[rr_n] \{\
.   nr rr_ip1 \\n[rr_i]+1
.   nr rr_count 0
.   if \\n[rr_levels!\\n[rr_ip1]]>\\n[rr_levels!\\n[rr_i]] \{\
.       nr rr_j \\n[rr_i] 1
.       while \\n+[rr_j]<\\n[rr_n] \{\
.           if \\n[rr_levels!\\n[rr_j]]<=\\n[rr_levels!\\n[rr_i]] \{\
.               break
.           \}
.           if \\n[rr_levels!\\n[rr_j]]==(\\n[rr_levels!\\n[rr_i]]+1) \{\
.               nr rr_count \\n[rr_count]+1
.           \}
.       \}
.   \}
.   ds hhh \\*[rr_labels!\\n[rr_i]]
.   pdfbookmark -\\n[rr_count] Link\\n[rr_i] "\\*[hhh]"
.\}
..
'\"
'\" Some postscript to make pdfmarks harmless on old interpreters...
'\"
.specialps "def /pdfmark where {pop} {userdict /pdfmark /cleartomark load put} ifelse"
'\"
'\" Force display of Bookmarks in Acrobat when document is viewed.
'\"
.pdfmark "[/PageMode /UseOutlines /Page 1 /View [/XYZ null null null] /DOCVIEW"
'\"
'\" Output the document info in pdfmarks
'\"
.pdfmark "\
	/Title (\*[an-title](\*[an-section])) \
	/Subject (\*[an-title] Manual Page) \
	/Author (Rick Richardson) \
	/Keywords (printing) \
	/Creator (groff \n(.x.\n(.y.\n(.Y -man) \
	/CreationDate (\*[an-extra1]) \
	/ModDate (\*[an-extra1]) \
	/DOCINFO"
\}
'\"
'\" The manual page name is only 1st level mark
'\"
.nr rr_levels!\n+[rr_n] 1
.ds rr_labels!\n[rr_n] \*[an-title](\*[an-section])
.pdfdest Link\n[rr_n]
'\"
'\"==========================================================================
'\"	MANUAL PAGE SOURCE
'\"==========================================================================
.SH NAME
printer-profile \- Profile using X-Rite ColorMunki and Argyll CMS
.SH SYNOPSIS
.B printer-profile
.RI [ options "] manuf model [rgb|cmyk] [patches] [ink-limit]"
.SH DESCRIPTION
.B printer-profile
prints a test chart, uses the ColorMunki instrument to scan it in, then
computes an ICM profile using the Argyll Color Management System.

Manuf is "sam".  Model is "clp-300", "clp-315", "clp-325" or "clp-365".
.br
Manuf is "hp".  Model is "2600" or "cp1215".
.br
Manuf is "km".  Model is "2300" or "2530".
.br
Manuf is "dell". Model is "1355".
.br

"rgb" is the usual setting.  "patches" is a multiple 196 per page.

Edit the script for additional models.

.SH OPTIONS
    -b 1|2        Bits per pixel (1)
    -r XRESxYRES  Resolution. Default=''. ()
    -P rem-print  Remote print (64-bit) machine, or none (amd)
    -S rem-scan   Remote scan (ColorMunki) machine, or none (mac)
    -D lvl        Debug level

.SH EXAMPLES
Profile the Samsung clp-315:

.nf
.ft CW
$ printer-profile sam 315 rgb 196
.ft P
.fi

.SH BUGS
gs 8.64 and before has problems with 32-bit machines and color profile
data.  Don't use!

You need Argyll_V1.5.1 or later.
.SH FILES
.BR /usr/bin/printer-profile,
.BR /usr/share/foo2*/icm/testing.icm
.SH SEE ALSO
firefox http://www.xritephoto.com/html/colormunkisplash.htm
.br
firefox http://www.argyllcms.com/
.br
firefox http://www.argyllcms.com/Argyll_V1.5.1_src.zip
.SH "AUTHOR"
Rick Richardson <rick.richardson@comcast.net>
.br
http://foo2zjs.rkkda.com/
'\"
'\"
'\"
.em pdf_outline


ChangeLog
2020-06-10	Rick Richardson <rick.richardson@comcast.net>
	* osx-hotplug: Add PREFIX=/usr or PREFIX=/usr/local to cc rules

2020-05-05:	Rick Richardson <rick.richardson@comcast.net>
	* foo2hbpl2.c: SaveToner implemented for color pages,
	  thanks to bedi@3mtech.sk .

2020-04-26:	Rick Richardson <rick.richardson@comcast.net>
	* foo2hbpl2.c: Bug with SaveToner

2020-02-07:	Rick Richardson <rick.richardson@comcast.net>
	* Add Cassette 2 to foo2oak-z1-InputSlot.xml (KM-1635)

2019-09-09:	Rick Richardson <rick.richardson@comcast.net>
	* Add SmartInstallDisable-Tool.run since it was gone from
	      http://hplipopensource.com/hplip-web/smartinstall/
	  See HP SMART INSTALL NOTES in the INSTALL doc.

2019-04-13:	Rick Richardson <rick.richardson@comcast.net>
	* Makefile: Add install-aa to disable AppArmor on SUSE

2018-09-10:	Rick Richardson <rick.richardson@comcast.net>
	* Makefile: target ppd:	revert the 09-06 change.

2018-09-06:	Rick Richardson <rick.richardson@comcast.net>
	* Makefile: target ppd:
	    Use the older foomatic-ppdfile which only has postscript,
	    not postscript and pdf, for making ppd's.

2017-12-02	Rick Richardson <rick.richardson@comcast.net>
	* New Printers: HP LaserJet Pro M12a, M12w

2017-03-20	Rick Richardson <rick.richardson@comcast.net>
	* Add ddstdecode, for decoding some Ricoh printers.

2016-10-21	Rick Richardson <rick.richardson@comcast.net>
	* Makefile: install-hotplug-prog
	    Add /lib/systemd/systemd-udevd to the list for Ubuntu 16.

2016-09-02	Rick Richardson <rick.richardson@comcast.net>
	* New Printer: Xerox Phaser 3040 using the foo2hbpl2 driver.

2016-08-30	Rick Richardson <rick.richardson@comcast.net>
	* New Printer: Xerox Phaser 3010 using the foo2hbpl2 driver.

2016-03-13	Rick Richardson <rick.richardson@comcast.net>
	* icc2ps/lcms.h: USE_CUSTOM_SWAB defined.

2015-10-24	Rick Richardson <rick.richardson@comcast.net>
	* foomatic-db/printer/Epson-AcuLaser_CX17NF.xml:
		Change margins on Epson AcuLaser CX17NF.

2015-10-11	Rick Richardson <rick.richardson@comcast.net>
	* New Printer: Epson AcuLaser CX17NF using the foo2hbpl2 driver.

2015-07-04	pavel.shvagirev	<pavel.shvagirev@gmail.com>
	* Modify: hplj10XX to not call us twice on Ubuntu 14.

2015-05-11	Rick Richardson <rick.richardson@comcast.net>
	* Bug in foomatic-db/printer/Xerox-Phaser_6115MFP.xml . Should use
	  foo2lava driver. Thanks to user "ophers".

2015-02-11	Rick Richardson <rick.richardson@comcast.net>
	* New Printer: Oki C511 using foo2hiperc.

2014-12-21	Rick Richardson <rick.richardson@comcast.net>
	* osx-hotplug: make it work with OS X 10.10

2014-12-04	Rick Richardson <rick.richardson@comcast.net>
	* make cups: add "Sandboxing Relaxed" to cups-files.conf iff cups >= 2

2014-09-25	Rick Richardson <rick.richardson@comcast.net>
	* foo2hblp2, foo2hblp2-wrapper, foomatic et al:
	    Add paper size A5.
	* foo2hblp2 foomatic db entries:
	    Changed resolution to 600x600, 1200x600, and 1200x1200.
	    Note: need to recreate the printers if you want more choices!

2014-05-19	Rick Richardson <rick.richardson@comcast.net>
	* HP CLJ CP1215:  Adjust margins: *HWMargins: 18 14.40 18 14

2014-05-04	Rick Richardson <rick.richardson@comcast.net>
	* foo2hbpl2-wrapper: Change PAPER=21 to PAPER=0 for custom page sizes.

2014-04-17	Rick Richardson <rick.richardson@comcast.net>
	* New Printer: Fuji Xerox DocuPrint CM215 MFP using the foo2hbpl2
	driver.

2014-04-08	Rick Richardson <rick.richardson@comcast.net>
	* Upgrade to JBIGKIT 2.1

2014-04-01	Rick Richardson <rick.richardson@comcast.net>
	* foo2hbpl2 foomatic-db files and PPDs:
	    Bug with default PageSize.
	    PLEASE delete and recreate the foo2hbpl2 printers.

2014-03-22	Rick Richardson <rick.richardson@comcast.net>
	* hbpldecode: another redundancy from Dave Coffin

2014-03-18	Rick Richardson <rick.richardson@comcast.net>
	* hbpldecode: More envelope sizes.

2014-03-17	Rick Richardson <rick.richardson@comcast.net>
	* foo2hbpl2.c: Bug, was using A4(1) instead of letter(4) by default
	* hbpldecode: Print out paper codes.

2014-03-15	Rick Richardson <rick.richardson@comcast.net>
	* hbpldecode: More changes from Dave.  Still gets 
	    Segmentation fault (core dumped)
	* hbpldecode: Changed unsigned to int in 2 places.  Works good!!!

2014-03-13	Rick Richardson <rick.richardson@comcast.net>
	* hbpldecode: Bug with no -d/-r.

2014-03-13	Rick Richardson <rick.richardson@comcast.net>
	* hbpldecode: Changes to pick apart version 1 from
	    Dave Coffin <dcoffin@cybercom.net>
	* I get:
	    Decoding page 1 to page-01.ppm ...
	    Segmentation fault (core dumped)

2014-03-02	Rick Richardson <rick.richardson@comcast.net>
	* New Printer: Fuji Xerox DocuPrint M215 MFP using the foo2hbpl2 driver.

2014-02-09	Rick Richardson <rick.richardson@comcast.net>
	* New Printer: Xerox WorkCentre 3045 MFP using the foo2hbpl2 driver.

2014-01-26	Rick Richardson <rick.richardson@comcast.net>
	* foo2qpdl et al: with -z3 (e.g. CLP-365) you can adjust
	    Brightness and Contrast for monochrome prints, and you can adjust
	    Saturation, Cyan-Red, Magenta-Green, Yellow-Blue Balances.
	* foo2hp: somewhere a bug in foomatic (probably 4.x) crept in
	    so foo2hp-AlignCMYK.xml was only processing one option instead
	    of three options!  Split the file into foo2hp-AlignC.xml,
	    foo2hp-AlignM.xml, and foo2hp-AlignY.xml.

2014-01-24	Rick Richardson <rick.richardson@comcast.net>
	* New Printer: Dell C1765nf/nfw using the foo2hbpl2 driver.

2014-01-08	Rick Richardson <rick.richardson@comcast.net>
	* New Printer: Epson AcuLaser M1400 using the foo2hbpl2 driver.

2013-12-25	Rick Richardson <rick.richardson@comcast.net>
	* New Printer: HP LaserJet Pro M1212nf MFP using the foo2xqx driver.

2013-12-01	Rick Richardson <rick.richardson@comcast.net>
	* foo2hiperc: clean up man pages

2013-11-29	Rick Richardson <rick.richardson@comcast.net>
	* foo2hiperc-wrapper, foomatic and PPD stuff:
	    add -z1 for Oki C810 so everything will be

printer-profile.sh
#!/bin/bash

PROGNAME="$0"

usage() {
	cat <<EOF
NAME
    `basename $PROGNAME` - printer-profile using X-Rite ColorMunki and Argyll CMS

SYNOPSIS
    `basename $PROGNAME` [options] manuf model [rgb|cmyk] [patches] [ink-limit]

DESCRIPTION
    Prints a test chart, uses the ColorMunki instrument to scan it in, then
    computes an ICM profile using the Argyll Color Management System.

    Manuf is "sam".  Model is "clp-300", "clp-315", "clp-325" or "clp-365".
    Manuf is "hp".  Model is "2600" or "cp1215" or "cp1025".
    Manuf is "km".  Model is "1600" or "2300" or "2530".
    Manuf is "dell". Model is "1355" or "1765"

    "rgb" is the usual setting.  "patches" is a multiple 196 per page.

    Edit the script for additional models.

OPTIONS
    -b 1|2        Bits per pixel ($BPP)
    -g gs-bin     Set ghostscript to "gs-bin", (gs)
    -r XRESxYRES  Resolution. Default=''. ($RES)
    -P rem-print  Remote print (64-bit) machine, or none ($REMPRINT)
    -S rem-scan   Remote scan (ColorMunki) machine, or none ($REMSCAN)
    -D lvl        Debug level

EXAMPLE
    $ printer-profile sam 315 rgb 196

BUGS
    gs 8.64 and before has problems with 32-bit machines and color profile
    data.  Don't use!

    You need Argyll_V1.6.3 or later in $ARGYLL_BIN.

SEE ALSO
    http://www.xritephoto.com/html/colormunkisplash.htm
	from Amazon, \$390 shipped
    http://www.argyllcms.com/
	free!
    http://www.argyllcms.com/Argyll_V${ARGYLL_VER}_src.zip
EOF

	exit 1
}

#
#       Report an error and exit
#
error() {
	echo "`basename $PROGNAME`: $1" >&2
	exit 1
}

debug() {
	if [ $DEBUG -ge $1 ]; then
	    echo "`basename $PROGNAME`: $2" >&2
	fi
}

#
#	Execute a command as root
#
root() {
    if [ -x /usr/bin/root ]; then
        /usr/bin/root $@
    else
        su -c "$*"
    fi
}

#
#	trap on error
#
trap "exit 1" ERR

#
#       Process the options
#
ARGYLL_VER=1.3.2
ARGYLL_VER=1.3.3
ARGYLL_VER=1.3.4
ARGYLL_VER=1.3.5
ARGYLL_VER=1.5.0
ARGYLL_VER=1.5.1
ARGYLL_VER=1.6.3
ARGYLL_ROOT=$HOME/src/Argyll_V${ARGYLL_VER}
ARGYLL_REF=$ARGYLL_ROOT/ref
ARGYLL_BIN=$ARGYLL_ROOT/bin
export PATH=$ARGYLL_BIN:$PATH

REMPRINT=amd
REMPRINT=none
REMSCAN=mac
REMSCAN=none
RGB=rgb
PATCHES=196	#Per page!
BPP=1
RES=
DEBUG=0
while getopts "b:g:r:P:S:D:h?" opt
do
	case $opt in
	b)	BPP="$OPTARG";;
	g)	export GSBIN="$OPTARG";;
	r)	RES="$OPTARG";;
	P)	REMPRINT="$OPTARG";;
	S)	REMSCAN="$OPTARG";;
	D)	DEBUG="$OPTARG";;
	h|\?)	usage;;
	esac
done
shift `expr $OPTIND - 1`

if [ ! -x $ARGYLL_BIN/printtarg ]; then
    error "No Argyll bin in $ARGYLL_BIN!"
fi

#
#	Reference ICM for colprof
#
reficm=$ARGYLL_REF/sRGB.icm
if [ ! -r $reficm ]; then
    error "No ref. icm in '$reficm'"
fi

#
#	Main Program
#
if [ $# -lt 2 ]; then
    usage
fi

MANUF="$1"
MODEL="$2"
if [ $# -ge 3 ]; then
    RGB="$3"
fi
if [ $# -ge 4 ]; then
    if [ "$4" -lt 10 ]; then
	PATCHES=`expr $4 \* 196`
    else
	PATCHES="$4"
    fi
fi

if [ $# -ge 5 ]; then
    INK="$5"
else
    INK=250
fi

BPP_b="-b$BPP"
RES_r=
if [ "$RES" != "" ]; then
    RES_r="-r$RES"
fi

case "$MANUF" in
km)
    MANUF=km
    case "$MODEL" in
    *1600*)
	FOO=foo2lava
	WRAPPER="foo2lava-wrapper $RES_r -z2 -c -C10 -Gnone.icm"
	OUT="root cp xxx.prn /dev/usb/lp1"
	;;
    *2530*)
	FOO=foo2lava
	WRAPPER="foo2lava-wrapper $RES_r -z0 -c -C10 -Gnone.icm"
	OUT="nc 192.168.1.13 9100 < xxx.prn"
	;;
    *2300*)
	FOO=foo2zjs
	WRAPPER="foo2zjs-wrapper $RES_r -c -C10 -Gnone.icm"
	OUT="nc 192.168.1.10 9100 < xxx.prn"
	;;
    *)
	usage
	;;
    esac
    ;;
sam*)
    MANUF=sam
    FOO=foo2qpdl
    case "$MODEL" in
    *300*)
	WRAPPER="foo2qpdl-wrapper $RES_r -z0 -c -C10 -Gnone.icm"
	OUT="nc 192.168.1.11 9100 < xxx.prn"
	;;
    *315*|*325*)
	WRAPPER="foo2qpdl-wrapper $RES_r -z2 -c -C10 -Gnone.icm"
	OUT="root cp xxx.prn /dev/usb/lp1"
	;;
    *365*)
	WRAPPER="foo2qpdl-wrapper $RES_r -z3 -c -C10 -Gnone.icm"
	OUT="root cp xxx.prn /dev/usb/lp1"
	;;
    *)
	usage
	;;
    esac
    ;;
hp*)
    MANUF=hp
    case "$MODEL" in
    *2600*)
	FOO=foo2hp
	WRAPPER="foo2hp2600-wrapper $RES_r $BPP_b -z0 -c -C10 -Gnone.icm"
	OUT="nc 192.168.1.12 9100 < xxx.prn"
	;;
    *cp1215*)
	FOO=foo2hp
	WRAPPER="foo2hp2600-wrapper $RES_r $BPP_b -z1 -c -C10 -Gnone.icm"
	OUT="root cp xxx.prn /dev/usb/lp2"
	;;
    *cp1025*)
	FOO=foo2zjs
	WRAPPER="foo2zjs-wrapper $RES_r -z3 -c -C10 -Gnone.icm"
	OUT="root cp xxx.prn /dev/usb/lp2"
	OUT="nc 192.168.1.16 9100 < xxx.prn"
	;;
    *)
	usage
	;;
    esac
    ;;
dell*)
    MANUF=dell
    case "$MODEL" in
    *1355*|*1765*)
        FOO=foo2hbpl2
        WRAPPER="foo2hbpl2-wrapper $RES_r -c -C10 -Gnone.icm"
        OUT="nc 192.168.178.41 9100 < xxx.prn"
        ;;
    *)
        usage
        ;;
    esac
    ;;
*)
    usage
    ;;
esac

mrp="$MANUF-$MODEL-$RGB-$PATCHES"
if [ "$INK" != "" ]; then
    mrp="$mrp-ink$INK"
fi
if [ "$BPP" != "" ]; then
    mrp="$mrp-bpp$BPP"
fi
if [ "$RES" != "" ]; then
    mrp="$mrp-$RES"
fi
echo "$mrp"

targen_opts=
case "$RGB" in
rgb|RGB)	targen_opts="$targen_opts -d2";;
cmyk|CMYK)	targen_opts="$targen_opts -d4";;
*)		error "Parm2: Must be rgb or cmyk";;
esac

targen_opts="$targen_opts -f$PATCHES"

if [ "$INK" != "" ]; then
    targen_opts="$targen_opts -l$INK"
fi

echo
echo "******************************** targen ********************************"
echo "targen $targen_opts $mrp"
if ! targen $targen_opts $mrp >$mrp.err1; then
    cat $mrp.err1
    exit 1
fi

echo
echo "******************************* printtarg ******************************"
# -h	Use double density for CM
# -v	Verbose mode
# -iCM	Select target instrument, CM = ColorMunki
# -pLetter
# -R0	Use given random start number
echo "$ARGYLL_BIN/printtarg -h -v -iCM -p Letter -R0 $mrp"
$ARGYLL_BIN/printtarg -h -v -iCM -p Letter -R0 $mrp

evince $mrp.ps &

echo
echo "******************************* print it *******************************"
echo "$WRAPPER"
case "$REMPRINT" in
''|none)
    $WRAPPER <$mrp.ps >$mrp.prn
    ;;
*)
    ssh $REMPRINT "$WRAPPER" <$mrp.ps >$mrp.prn
    ;;
esac
ls -l $mrp.prn
echo -n "Print it? [y|n]? "
read yes
case "$yes" in
y|Y)
    cp $mrp.prn xxx.prn
    echo "	$OUT"
    eval $OUT
    rm -f xxx.prn
esac

echo
echo "******************************* chartread ******************************"
case "$REMSCAN" in
''|none)
    echo "$ARGYLL_BIN/chartread $mrp"
    $ARGYLL_BIN/chartread $mrp
    ;;
*)
    echo "scp $mrp.ti2 $REMSCAN:"
    scp $mrp.ti2 $REMSCAN:
    echo "$ARGYLL_BIN/chartread $mrp"
    echo  -n "Scanned on remote system $REMSCAN? [y/n]? "
    read yes
    case "$yes" in
    y|Y)
	scp $REMSCAN:$mrp.ti3 .
	;;
    esac
    ;;
esac

echo
echo "******************************** colprof *******************************"
case $RGB in
rgb)
    echo "colprof -v -D\"$mrp\" -S $reficm -qm -cmt -dpp $mrp"
    colprof -v -D"$mrp" -S $reficm -qm -cmt -dpp $mrp
    ;;
cymk)
    echo "colprof -v -D\"$mrp\" -S $reficm -qm  -cmt -dpp -kr $mrp"
    colprof -v -D"$mrp" -S $reficm -qm  -cmt -dpp -kr $mrp
    ;;
esac

mv $mrp.icc $mrp.icm
root cp $mrp.icm /usr/share/$FOO/icm/testing.icm
echo "/usr/share/$FOO/icm/testing.icm created!"
ls -l /usr/share/$FOO/icm/
